sematic/types/registry.py

```diff
@@ -300,6 +322,10 @@ def assert_supported(type_):
                 "'Any' is not a Sematic-supported type. Use 'object' instead."
             )
 
+        if _is_type(type_):
+            for assertion in _TYPE_ASSERTIONS:
+                assertion(type_)
```

Comment:
> Why not just use an if for ExternalResource for now

Because it seems like it would be weird for the type registry to depend on `external_resource`. The type registry is a pretty fundamental module that is depended on directly or indirectly by a lot of stuff. It would be very easy to accidentally introduce circular dependencies if it depended on something as "downstream" as `external_resource`.

Will put together a vote setup, but won't convert the existing checks into votes for now.

Reply:
discussed offline; will use a local import of `ExternalResource` and check that way